From 8e698df619d12618a75fc1a016cb06d87ebfb764 Mon Sep 17 00:00:00 2001
From: Michal Frynas <michal.frynas@inter.ikea.com>
Date: Thu, 20 Feb 2025 14:55:28 +0100
Subject: [PATCH] Adjust-meson.build-to-meson-higher-then.0.60

---
 meson.build | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/meson.build b/meson.build
index 1bdb9c7b32..48d18d8c84 100644
--- a/meson.build
+++ b/meson.build
@@ -38,7 +38,7 @@ conf.set_quoted('RELATIVE_SOURCE_PATH', relative_source_path)
 want_ossfuzz = get_option('oss-fuzz')
 want_libfuzzer = get_option('llvm-fuzz')
 want_fuzzbuzz = get_option('fuzzbuzz')
-if want_ossfuzz + want_libfuzzer + want_fuzzbuzz > 1
+if want_ossfuzz and want_libfuzzer and want_fuzzbuzz > 1
         error('only one of oss-fuzz, llvm-fuzz or fuzzbuzz can be specified')
 endif
 
-- 
2.40.1

