From 5cf54778bdc9b9b807603b2824fed25b0b401933 Mon Sep 17 00:00:00 2001
From: Michal Frynas <michal.frynas@sigma.se>
Date: Thu, 30 Sep 2021 15:33:58 +0200
Subject: [PATCH] Disabling environment save in U-Boot triggers incorrect error

---
 include/configs/stm32mp1.h | 7 ++++++
 1 file changed, 6 insertions(+)

diff --git a/include/configs/stm32mp1.h b/include/configs/stm32mp1.h
index 8fa3322093..104a4f29c0 100644
--- a/include/configs/stm32mp1.h
+++ b/include/configs/stm32mp1.h
@@ -147,6 +147,12 @@
  * 1M fdt, 1M script, 1M pxe and 1M for splashimage
  * and the ramdisk at the end.
  */
+#ifdef CONFIG_CMD_SAVEENV
+#define SAVE_DEFAULT " then env set env_default 0;env save;fi\0"
+#else
+#define SAVE_DEFAULT " then env set env_default 0;fi\0"
+#endif
+
 #define CONFIG_EXTRA_ENV_SETTINGS \
 	"kernel_addr_r=0xc2000000\0" \
 	"fdt_addr_r=0xc4000000\0" \
@@ -156,6 +161,7 @@
 	"ramdisk_addr_r=0xc4400000\0" \
 	"altbootcmd=echo 'Boot B';setenv altboot 1;run bootcmd_stm32mp\0" \
 	"env_check=if env info -p -d -q; then env save; fi\0" \
+	SAVE_DEFAULT \
 	STM32MP_BOOTCMD \
 	BOOTENV \
 	"boot_net_usb_start=true\0"
-- 
2.27.0

